encoding/encodeInto.any.js

```diff
@@ -73,77 +73,91 @@
+              view = new Uint8Array(buffer, destinationOffset, destinationLength),
+              fullView = new Uint8Array(buffer),
+              control = new Array(bufferLength);
+        let byte = destinationFiller;
+        for (let i = 0; i < bufferLength; i++) {
+          if (destinationFiller === "random") {
+            byte = Math.floor(Math.random() * 256);
+          }
+          control[i] = byte;
+          fullView[i] = byte;
         }
-        control[i] = byte;
-        fullView[i] = byte;
-      }
-
-      // It's happening
-      const result = encoder.encodeInto(testData.input, view);
-
-      // Basics
-      assert_equals(view.byteLength, destinationLength);
-      assert_equals(view.length, destinationLength);
-
-      // Remainder
-      assert_equals(result.read, testData.read);
-      assert_equals(result.written, testData.written.length);
-      for (let i = 0; i < bufferLength; i++) {
-        if (i < destinationOffset || i >= (destinationOffset + testData.written.length)) {
-          assert_equals(fullView[i], control[i]);
-        } else {
-          assert_equals(fullView[i], testData.written[i - destinationOffset]);
+  
+        // It's happening
+        const result = encoder.encodeInto(testData.input, view);
+  
+        // Basics
+        assert_equals(view.byteLength, destinationLength);
+        assert_equals(view.length, destinationLength);
+  
+        // Remainder
+        assert_equals(result.read, testData.read);
+        assert_equals(result.written, testData.written.length);
+        for (let i = 0; i < bufferLength; i++) {
+          if (i < destinationOffset || i >= (destinationOffset + testData.written.length)) {
+            assert_equals(fullView[i], control[i]);
+          } else {
+            assert_equals(fullView[i], testData.written[i - destinationOffset]);
+          }
         }
-      }
-    }, "encodeInto() with " + testData.input + " and destination length " + testData.destinationLength + ", offset " + destinationData.destinationOffset + ", filler " + destinationData.filler);
+      }, "encodeInto() into "  + ArrayBufferOrSharedArrayBuffer.name + " with " + testData.input + " and destination length " + testData.destinationLength + ", offset " + destinationData.destinationOffset + ", filler " + destinationData.filler);
+    })
   });
 });
 
-[DataView,
- Int8Array,
- Int16Array,
- Int32Array,
- Uint16Array,
- Uint32Array,
- Uint8ClampedArray,
- Float32Array,
- Float64Array].forEach(view => {
-  test(() => {
-    assert_throws(new TypeError(), () => new TextEncoder().encodeInto("", new view(new ArrayBuffer(0))));
-  }, "Invalid encodeInto() destination: " + view.name);
- });
-
-test(() => {
-  assert_throws(new TypeError(), () => new TextEncoder().encodeInto("", new ArrayBuffer(10)));
-}, "Invalid encodeInto() destination: ArrayBuffer");
+[ArrayBuffer, self["SharedArrayBuffer"]].forEach((ArrayBufferOrSharedArrayBuffer) => {
```

Comment:
You could also wrap this forEach around the first two tests avoiding the duplication of the typed arrays forEach.

Reply:
Done